src/test/mina_automation/daemon.ml

```diff
@@ -0,0 +1,186 @@
+(**
+Module to run daemon process.
+*)
+open Core
+
+open Async
+
+module Paths = struct
+  let dune_name = "src/app/cli/src/mina.exe"
+
+  let official_name = "mina"
+end
+
+module Executor = Executor.Make (Paths)
+
+type t = Executor.t
+
+let logger = Logger.create ()
+
+(** 
+  Module [Client] provides functions to interact with a Mina daemon.
+*)
+module Client = struct
+  type t = { port : int; executor : Executor.t }
+
+  let create ?(port = 3085) ?(executor = Executor.AutoDetect) () =
+    { port; executor }
+
+  (** [stop_daemon t] stops the daemon running on the specified port.
+    @param t The daemon instance containing the executor and port information.
+    @return Unit. Executes the command to stop the daemon using the executor.
+  *)
+  let stop_daemon t =
+    Executor.run t.executor
+      ~args:[ "client"; "stop-daemon"; "-daemon-port"; sprintf "%d" t.port ]
+      ()
+
+  (** [daemon_status t] retrieves the status of the daemon running on the specified port.
+    It executes the command `client status -daemon-port <port>` using the provided executor.
+    
+    @param t The daemon instance containing the executor and port information.
+    @return The result of the executor run, which may be ignored if the command fails.
+  *)
+  let daemon_status t =
+    Executor.run t.executor
+      ~args:[ "client"; "status"; "-daemon-port"; sprintf "%d" t.port ]
+      ~ignore_failure:true ()
+
+
+  (** [wait_for_bootstrap t ?client_delay ?retry_delay ?retry_attempts ()] waits for the daemon to bootstrap.
+    @param t The daemon instance containing the executor and port information.
+    @param client_delay The delay before connecting to the daemon.
+    @param retry_delay The delay between retries.
+    @param retry_attempts The number of retries.
+    @return A deferred result indicating the success or failure of the operation. 
+  *)
+  let wait_for_bootstrap t ?(client_delay = 40.) ?(retry_delay = 60.)
+      ?(retry_attempts = 10) () =
+    Async.printf "Waiting initial %d s. before connecting\n"
+      (int_of_float client_delay) ;
+    let%bind _ =
+      Deferred.map (after @@ Time.Span.of_sec client_delay) ~f:Or_error.return
+    in
+    let rec go retries_remaining =
```

Comment:
I believe retry mechanism is everywhere in our codebase. Would be nice if we could factor out a module for retry/backing-off algorithm. But I understand it's bigger than the scope of this PR.

Reply:
Yes, i will not it down and fix it in another PR